docs/source/llms/langchain/autologging.rst

```diff
@@ -0,0 +1,257 @@
+                parent_run_id=parent_run_id,
+                span_type=SpanType.CHAT_MODEL,
+                run_id=run_id,
+                inputs=messages,
+                attributes=kwargs,
+            )
+
+Patch Functions for Logging Artifacts
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Other artifacts such as models, datasets, are logged by patching the invocation functions of the supported models to insert the logging call. MLflow patches the following functions:
+
+* ``invoke``
+* ``batch``
+* ``stream``
+* ``get_relevant_documents`` (for retrievers)
+* ``__call__`` (for Chains and AgentExecutors)
+
+
+Troubleshooting
+---------------
+
+If you encounter any issues with LangChain autologging, please refer to `FAQ <../index.html#faq>` first. If you still have questions, please feel free to open an issue in `MLflow Github repo <https://github.com/mlflow/mlflow/issues>`_.
+
+
+Documentation for Old Versions
+------------------------------
+
+MLflow LangChain Autologging feature is largely renewed in ``MLflow 2.14.0``. If you are using the earlier version of MLflow, please refer to following documentation.
+
+.. note::
+    To use MLflow LangChain autologging, please upgrade langchain to **version 0.1.0** or higher.
+    Depending on your existing environment, you may need to manually install langchain_community>=0.0.16 in order to enable the automatic logging of artifacts and metrics. (this behavior will be modified in the future to be an optional import)
+    If autologging doesn't log artifacts as expected, please check the warning messages in `stdout` logs. 
+    For langchain_community==0.0.16, you will need to install the `textstat` and `spacy` libraries manually, as well as restarting any active interactive environment (i.e., a notebook environment). On Databricks, you can achieve this via executing `dbutils.library.restartPython()` to force the Python REPL to restart, allowing the newly installed libraries to be available.
+
+MLflow langchain autologging injects `MlflowCallbackHandler <https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/callbacks/mlflow_callback.py>`_ into the langchain model inference process to log
+metrics and artifacts automatically. We will only log the model if both `log_models` is set to `True` when calling :py:func:`mlflow.langchain.autolog` and the objects being invoked are within the supported model types: `Chain`, `AgentExecutor`, `BaseRetriever`, `RunnableSequence`, `RunnableParallel`, `RunnableBranch`, `SimpleChatModel`, `ChatPromptTemplate`,
+`RunnableLambda`, `RunnablePassthrough`. Additional model types will be supported in the future.
+
+.. note::
+    We patch `invoke` function for all supported langchain models, `__call__` function for Chains, AgentExecutors models, and `get_relevant_documents` function for BaseRetrievers, so only when those functions are called MLflow autologs metrics and artifacts.
```

Comment:
```suggestion
    We patch the `invoke` function for all supported langchain models, the `__call__` function for Chains and AgentExecutors models, and `get_relevant_documents` function for BaseRetrievers so that only when those functions are called will MLflow automatically log metrics and artifacts.
```